### PR TITLE
add k8s-ro-volume.bats

### PR DESCRIPTION
**Pull request by Tim-0731-Hzt**
add k8s-ro-volume.bats

Fixes:#5337
Signed-off-by: Zhongtao Hu <zhongtaohu.tim@linux.alibaba.com>

### CODE REVIEW

File: integration/kubernetes/run_kubernetes_tests.sh [hunk 1]
```diff
@@ -95,6 +95,7 @@ else
                 "k8s-shared-volume.bats" \
                 "k8s-volume.bats" \
                 "k8s-nginx-connectivity.bats" \
+				"k8s-ro-volume.bats" \
```
Comment (littlejawa) [inline integration/kubernetes/run_kubernetes_tests.sh:98]: nit: the line is not aligned. Can you replace the tabs with spaces?

Comment (littlejawa) [inline integration/kubernetes/run_kubernetes_tests.sh:98]: Also: I think you need to remove "k8s-ro-volume.bats" from the comment above this "if" block.

Comment (Tim-0731-Hzt) [reply integration/kubernetes/run_kubernetes_tests.sh]: fixed

Comment (Tim-0731-Hzt) [reply integration/kubernetes/run_kubernetes_tests.sh]: fixed

**Review by littlejawa — APPROVED** [review]
lgtm
Thanks!

**Review by snir911 — APPROVED** [review]

File: integration/kubernetes/k8s-ro-volume.bats [hunk 2]
```diff
@@ -36,12 +36,19 @@ setup() {
 	sudo ls -lR $host_mounts/ | grep $tmp_file && echo "should not find $tmp_file in host mounts" && false
```
Comment (studychao) [inline integration/kubernetes/k8s-ro-volume.bats:36]: seems like `shared_mounts` and `host_mounts` are not valid in runtime-rs, is the check here also needed to be changed?

**Review by studychao — COMMENTED** [review]
Thanks Tim

File: integration/kubernetes/k8s-ro-volume.bats (hunk 2 above)
Comment (Tim-0731-Hzt) [reply integration/kubernetes/k8s-ro-volume.bats:36]: Yes, I will fix that

File: integration/kubernetes/k8s-ro-volume.bats [hunk 3]
```diff
@@ -32,16 +32,33 @@ setup() {
...
-	sudo ls -lR $shared_mounts/ | grep $tmp_file && echo "should not find $tmp_file in shared mounts" && false
-	sudo ls -lR $host_mounts/ | grep $tmp_file && echo "should not find $tmp_file in host mounts" && false
 
-	# Validate readonly volume mount on the host
-	container_id=$(sudo -E crictl ps -q --state Running --name $container_name)
+	# runtime-rs host path
+	rw="/run/kata-containers/shared/sandboxes/$pod_id/rw/"
+	ro="/run/kata-containers/shared/sandboxes/$pod_id/ro/"
+	
+	# runtime host path
```
Comment (studychao) [inline integration/kubernetes/k8s-ro-volume.bats:40]: runtime-go could be clearer.

Comment (Tim-0731-Hzt) [reply integration/kubernetes/k8s-ro-volume.bats]: fixed

**Pull request merged**